integration/src/test/java/org/mule/test/integration/routing/outbound/UntilSuccessfulPropagationTestCase.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.test.integration.routing.outbound;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import org.junit.Test;
+import org.mule.runtime.core.api.event.CoreEvent;
+import org.mule.test.AbstractIntegrationTestCase;
+
+public class UntilSuccessfulPropagationTestCase extends AbstractIntegrationTestCase {
+
+  @Override
+  protected String getConfigFile() {
+    return "org/mule/test/integration/routing/outbound/until-successful-propagation.xml";
+  }
+
+  @Test
+  public void variablePropagationToErrorHandler() throws Exception {
+    assertPayloadContent(flowRunner("variablePropagationToErrorHandler").withPayload("message").run());
```

Comment:
use a parameterized test instead of repeating the test all over and just changing the flow name.